sympy/core/relational.py

```diff
@@ -302,18 +303,37 @@ def __new__(cls, lhs, rhs=0, **options):
             # If expressions have the same structure, they must be equal.
             if lhs == rhs:
                 return S.true
-
-            # If appropriate, check if the difference evaluates.  Detect
-            # incompatibility such as lhs real and rhs not real.
+            elif all(isinstance(i, BooleanAtom) for i in (rhs, lhs)):
+                return S.false
+
+            # check finiteness
+            fin = L, R = [i.is_finite for i in (lhs, rhs)]
+            if None not in fin:
+                if L != R:
+                    return S.false
+                if L is False:
+                    return S.true
+
+            # see if the difference evaluates
+            if not any(isinstance(i, BooleanAtom) or isinstance(i, ImmutableMatrix) or \
+            isinstance(i, MatrixSymbol) or isinstance(i, Tuple)for i in (lhs, rhs)):
+                n, d = (lhs - rhs).as_numer_denom()
```

Comment:
@smichr problem is this condition ```.as_numer_denom()```. All of above condition for this because ```Boolean```,```MatrixSymbol```,```ImmutableMatrix```,```Tuple``` makes error.